util/job_manager.rb

```diff
@@ -58,22 +58,22 @@ def perform
         @wrapper = wrapper
         log "avancement: #{message} #{avancement}"
         if avancement && cost
-          p = Result.get(self.uuid) || { 'graph' => [] }
-          p['graph'] = [] if !p.has_key?('graph')
-          p['csv'] = true if ask_restitution_csv
-          p['graph'] << { iteration: avancement, cost: cost, time: time }
-          p['result'] = solution if solution
+          p = Result.get(self.uuid) || { graph: [] }
+          p[:graph] ||= []
```

Comment:
If for some reason, in other part of the code if we ever do `p[:graph] = nil` it will be overwritten here. I think the previous way was more secure -- or an improved version of it `unless p.key?(:graph)`.

Reply:
I just realized : 

if `p[:graph] = nil` then we will generate and error 2 lines after, when trying to add current iteration. I guess the aim of this `p['graph'] = [] if !p.has_key?('graph')` is to ensure `p['graph']` is never nil. 

In this case my edit would be ok, what do you think ?

Reply:
Seems ok to me